### PR TITLE
Updates to C3 and SAS convection schemes

### PR DESCRIPTION
**Pull request by lisa-bengtsson**
*Brings out tuning parameters from the progsigma_calc.F90, which are used in the saSAS and C3 shallow and deep cumulus convection schemes. 
*Adds a limiter on area fraction to not exceed 0.95 after final tuning
*Updates the regression test used for gfsv17 tests, to include extracted parameters in the namelist. 

For RRFSv1 ensemble members using saSAS convection, workflow updates are needed.

Some justification for the PR can be found here: https://docs.google.com/presentation/d/11WVCFAb-JM628N2RyAWLqjxxrSovKU8AB72xBZJ1FfI/edit#slide=id.p

Addresses issue: https://github.com/ufs-community/ufs-weather-model/issues/1964

### CODE REVIEW

File: physics/cu_c3_driver.F90 [hunk 1]
```diff
@@ -279,6 +280,14 @@ subroutine cu_c3_driver_run(ntracer,garea,im,km,dt,flag_init,flag_restart,&
 !$acc end kernels
      endif
 
+
+     if(progsigma)then
```
Comment (dustinswales) [inline physics/cu_c3_driver.F90]: Should this piece be in the _init() phase? And only done once?

File: physics/cu_c3_driver.F90 [hunk 2]
```diff
@@ -92,14 +93,14 @@ subroutine cu_c3_driver_run(ntracer,garea,im,km,dt,flag_init,flag_restart,&
...
    integer, intent(in   ) :: ichoice_in,ichoicem_in,ichoice_s_in
    logical, intent(in   ) :: flag_init, flag_restart, do_mynnedmf
    logical, intent(in   ) :: flag_for_scnv_generic_tend,flag_for_dcnv_generic_tend, &
-        do_ca,progsigma
-   real (kind=kind_phys), intent(in) :: g,cp,fv,r_d,xlv,r_v
+        do_ca
+   real (kind=kind_phys), intent(in) :: g,cp,fv,r_d,xlv,r_v,betascu,betamcu,betadcu
    logical, intent(in   ) :: ldiag3d
-
+   logical, intent(inout) :: progsigma
```
Comment (dustinswales) [inline physics/cu_c3_driver.F90]: You changed from intent(in) to intent(inout) for progsigma, but I don't see this change below in the metadata?

File: physics/cu_c3_sh.F90 [hunk 3]
```diff
@@ -672,13 +676,13 @@ subroutine cu_c3_sh_run (                                            &
               dz=z_cup(i,k)-z_cup(i,k-1)
               ! cloud liquid water
               c1d(i,k)=c1_shal! 0. !.02*up_massdetr(i,k-1)
+              clw_all(i,k)=max(0.,qco(i,k)-trash)
```
Comment (dustinswales) [inline physics/cu_c3_sh.F90]: Change 0. to 0._kind_phys

**Review by dustinswales — CHANGES_REQUESTED** [review]
Looks good. A few small comments/suggestions.

File: physics/cu_c3_driver.F90 (hunk 2 above)
Comment (lisa-bengtsson) [reply physics/cu_c3_driver.F90]: It's in there, I had to expand the lines on github to see the difference, right above the new input field "cnx"

File: physics/cu_c3_driver.F90 (hunk 1 above)
Comment (lisa-bengtsson) [reply physics/cu_c3_driver.F90]: yes, good idea

File: physics/cu_c3_driver.F90 (hunk 2 above)
Comment (dustinswales) [reply physics/cu_c3_driver.F90]: I see now. It looks like you already had intent(inout) before. All square now.

Comment (lisa-bengtsson) [reply physics/cu_c3_driver.F90:103]: hm...no it should be new with this update, on my end it is highlighted in green, but I still had to expand it. I did run the full RT with this code, so I think we should be good with the "intents". I will move the check to the initial part.

Comment (lisa-bengtsson) [reply physics/cu_c3_driver.F90]: It is highlighted in green on my end, it should be new with this PR. I did run the full RT with these changes, so I think we should be good with the "intents".

Comment (lisa-bengtsson) [reply physics/cu_c3_driver.F90]: image

^It's sort of baked in with the new variable

Comment (dustinswales) [reply physics/cu_c3_driver.F90]: Ah yes. very sneaky

**Review by Qingfu-Liu — APPROVED** [review]

**Review by JongilHan66 — APPROVED** [review]

**Review by haiqinli — APPROVED** [review]

**Review by dustinswales — APPROVED** [review]

**Comment by FernandoAndrade-NOAA** [comment]
Testing for #[1941 ](https://github.com/ufs-community/ufs-weather-model/pull/1941) has completed successfully, please continue with the merge process.

**Comment by zach1221** [comment]
> Testing for #[1941 ](https://github.com/ufs-community/ufs-weather-model/pull/1941) has completed successfully, please continue with the merge process.

Copying @lisa-bengtsson and @grantfirl as well. Please feel free to merge this PR.

**Review by grantfirl — APPROVED** [review]

**Pull request merged**